src/Components/Pages.tsx

```diff
@@ -0,0 +1,128 @@
+  SceneTimeRange,
+} from '@grafana/scenes';
+import {
+  DRILLDOWN_URL_KEYS,
+  navigateToIndex,
+  PLUGIN_BASE_URL,
+  prefixRoute,
+  ROUTE_DEFINITIONS,
+  ROUTES,
+  SERVICE_URL_KEYS,
+  SLUGS,
+} from '../services/routing';
+import { PageLayoutType } from '@grafana/data';
+import { IndexScene } from './IndexScene/IndexScene';
+import { locationService } from '@grafana/runtime';
+
+function getServicesScene() {
+  const DEFAULT_TIME_RANGE = { from: 'now-15m', to: 'now' };
+  return new EmbeddedScene({
+    body: new IndexScene({
+      $timeRange: new SceneTimeRange(DEFAULT_TIME_RANGE),
+    }),
+  });
+}
+
+// Index page
+export function makeIndexPage() {
+  return new SceneAppPage({
+    // Top level breadcrumb
+    title: 'Logs',
+    url: prefixRoute(SLUGS.explore),
+    layout: PageLayoutType.Custom,
+    preserveUrlKeys: SERVICE_URL_KEYS,
+    routePath: prefixRoute(SLUGS.explore),
+    getScene: () => getServicesScene(),
+    drilldowns: [
+      {
+        routePath: ROUTE_DEFINITIONS.logs,
+        getPage: (routeMatch, parent) => makeBreakdownPage(routeMatch, parent, SLUGS.logs),
+        defaultRoute: true,
+      },
+      {
+        routePath: ROUTE_DEFINITIONS.labels,
+        getPage: (routeMatch, parent) => makeBreakdownPage(routeMatch, parent, SLUGS.labels),
+      },
+      {
+        routePath: ROUTE_DEFINITIONS.patterns,
+        getPage: (routeMatch, parent) => makeBreakdownPage(routeMatch, parent, SLUGS.patterns),
+      },
+      {
+        routePath: ROUTE_DEFINITIONS.fields,
+        getPage: (routeMatch, parent) => makeBreakdownPage(routeMatch, parent, SLUGS.fields),
+      },
+      {
+        routePath: '*',
+        getPage: () => makeRedirectPage(),
+      },
+    ],
+  });
+}
+
+// Redirect page back to index
+export function makeRedirectPage() {
+  return new SceneAppPage({
+    title: '',
+    url: PLUGIN_BASE_URL,
+    getScene: makeEmptyScene(),
+    hideFromBreadcrumbs: true,
+    routePath: '*',
+    $behaviors: [
+      () => {
+        navigateToIndex();
+      },
+    ],
+  });
+}
+
+function makeEmptyScene(): (routeMatch: SceneRouteMatch) => EmbeddedScene {
+  return () =>
+    new EmbeddedScene({
+      body: new SceneFlexLayout({
+        direction: 'column',
+        children: [],
+      }),
+    });
+}
+
+export function makeBreakdownPage(
+  routeMatch: SceneRouteMatch<{ service: string }>,
+  parent: SceneAppPageLike,
+  slug: SLUGS
+): SceneAppPage {
+  const { service } = extractServiceFromRoute(routeMatch);
+
+  return new SceneAppPage({
+    title: slugToBreadcrumbTitle(slug),
+    layout: PageLayoutType.Custom,
+    url: ROUTES[slug](service),
+    preserveUrlKeys: DRILLDOWN_URL_KEYS,
+    getParentPage: () => parent,
+    getScene: () => getServicesScene(),
+  });
+}
+
+export function getSlug() {
+  const location = locationService.getLocation();
+  const slug = location.pathname.slice(location.pathname.lastIndexOf('/') + 1, location.pathname.length);
+  return slug as SLUGS;
```

Comment:
I think we can do a bit better with this type. It's also unexpected to be all uppercase. Maybe `AppSection`, `AppViewType`, `LogsExplorationView` or something like this.

Reply:
Updated